WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostMoreButtonUiStateBuilder.kt

```diff
@@ -57,8 +57,8 @@ class ReaderPostMoreButtonUiStateBuilder @Inject constructor(
         menuItems.add(SpacerNoAction())
         menuItems.add(buildBlockSite(onButtonClicked))
         menuItems.add(buildReportPost(onButtonClicked))
-        menuItems.add(buildReportUser(onButtonClicked))
-        checkAndAddMenuItemForBlockUser(menuItems, onButtonClicked)
+        checkAndAddMenuItemForReportUser(post, menuItems, onButtonClicked)
+        checkAndAddMenuItemForBlockUser(post, menuItems, onButtonClicked)
```

Comment:
Ever-so-minor nitpicky comment:

I think this is fine as is, but we could also consider something like the following:

```kotlin
if (!readerUtilsWrapper.isSelfHosted(post.authorBlogId)) {
    menuItems.add(buildReportUser(onButtonClicked))
    menuItems.add(buildBlockUser(onButtonClicked))
}
```

Since the extracted functions are almost identical.. otoh, the way it is now is more consistent with the other extracted functions :man_shrugging: .

Maybe we can extract a single function: `checkAndAddUserMenuItems` :thinking:

Reply:
That's a great suggestion @mkevins 👍 

> the way it is now is more consistent with the other extracted functions

That's true but on a second thought I too feel that this is unnecessary boilerplate code 😅 
I've implemented your suggestion with 0a0afe81ed6478d070f773aac6fd07608025f9b0